include/ddc/kernels/splines/splines_linear_problem_dense.hpp

```diff
@@ -0,0 +1,113 @@
+ *
+ * The storage format is dense row-major. Lapack is used to perform every matrix and linear solver-related operations.
+ *
+ * @tparam ExecSpace The Kokkos::ExecutionSpace on which operations related to the matrix are supposed to be performed. Note: atm this is a placeholder for futur developments on GPU.
+ */
+template <class ExecSpace>
+class SplinesLinearProblemDense : public SplinesLinearProblem<ExecSpace>
+{
+public:
+    using typename SplinesLinearProblem<ExecSpace>::MultiRHS;
+    using SplinesLinearProblem<ExecSpace>::size;
+    using SplinesLinearProblem<ExecSpace>::operator<<;
+
+protected:
+    Kokkos::View<double**, Kokkos::LayoutRight, Kokkos::HostSpace> m_a;
+    Kokkos::View<int*, Kokkos::HostSpace> m_ipiv;
+
+public:
+    /**
+     * @brief SplinesLinearProblemDense constructor.
+     *
+     * @param mat_size The size of one of the dimensions of the square matrix.
+     */
+    explicit SplinesLinearProblemDense(const std::size_t mat_size)
+        : SplinesLinearProblem<ExecSpace>(mat_size)
+        , m_a("a", mat_size, mat_size)
+        , m_ipiv("ipiv", mat_size)
+    {
+        assert(mat_size > 0);
+
+        Kokkos::deep_copy(m_a, 0.);
+    }
+
+    virtual double get_element(std::size_t const i, std::size_t const j) const override
+    {
+        assert(i < size());
+        assert(j < size());
+        return m_a(i, j);
+    }
+
+    virtual void set_element(std::size_t const i, std::size_t const j, double const aij) override
+    {
+        m_a(i, j) = aij;
+    }
+
+    /**
+     * @brief Perform a pre-process operation on the solver. Must be called after filling the matrix.
+     *
+     * LU-factorize the matrix A and store the pivots using the LAPACK dgetrf() implementation.
+     */
+    void setup_solver() override
+    {
+        int const info = LAPACKE_dgetrf(
+                LAPACK_ROW_MAJOR,
+                size(),
+                size(),
+                m_a.data(),
+                size(),
+                m_ipiv.data());
+        if (info != 0) {
+            throw std::runtime_error("LAPACK failed with error code " + info);
+        }
+    }
+
+    /**
+     * @brief Solve the multiple right-hand sides linear problem Ax=b or its transposed version A^tx=b inplace.
+     *
+     * The solver method gaussian elimination with partial pivoting using the LU-factorized matrix A. The implementation is LAPACK method dgetrs.
+     *
+     * @param[in, out] multi_rhs A 2D Kokkos::View storing the multiple right-hand sides of the problem and receiving the corresponding solution.
+     * @param transpose Choose between the direct or transposed version of the linear problem.
+     */
+    void solve(MultiRHS b, bool const transpose) const override
+    {
+        assert(b.extent(0) == size());
+
+        auto b_host = create_mirror_view(Kokkos::DefaultHostExecutionSpace(), b);
+        Kokkos::deep_copy(b_host, b);
+        int const info = LAPACKE_dgetrs(
+                LAPACK_ROW_MAJOR,
```

Comment:
Remark from LAPACKE documentation:

"Note that using row-major ordering may require more memory and time than column-major ordering, because the routine must transpose the row-major order to the column-major order required by the underlying LAPACK routine."

So, there is an underlying CPU transposition involved here. Using LAPACK_ROW_MAJOR in dgetrs requires using row-major (LayoutRight) `b` AND `A`. So here matrix A is transposed at each call (which is unecessary because we never use it's non-transposed version).

Let me know if it suits the requirements.